test/integration/master_benchmark_test.go

```diff
@@ -0,0 +1,167 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/test/integration/framework"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"sync"
+	"testing"
+)
+
+const (
+	// Most of these benchmarks do something for b.N pods, so you generally
+	// just want an Iter of 1 unless you're looking to use more sophisticated
+	// pprof features like generating a callgraph.
+	Iter = 1
+
+	// Number of goroutines to balance b.N across = Parallelism * GORUNPROCS (set via -cpu)
+	// in tests that use RunParallel. This also controls the number of parallel goroutines
+	// started in other tests to do network bound io like apiserver requests.
+	Parallelism = 100
+)
+
+// If you need to start an etcd instance by hand, you also need to insert a key
+// for this check to pass (*any* key will do, eg:
+//curl -L http://127.0.0.1:4001/v2/keys/message -XPUT -d value="Hello world").
+func init() {
+	framework.RequireEtcd()
+}
+
+// Benchmark rc lifecycle
+func BenchmarkControllerCreate(b *testing.B) {
+	b.StopTimer()
+	m := framework.NewMasterComponents(&framework.Config{nil, true})
+	defer m.Stop()
+
+	data, err := ioutil.ReadFile("benchmark-controller.json")
+	if err != nil {
+		glog.Fatalf("Unexpected error: %v", err)
+	}
+	glog.Infof("Starting controller benchmark with N=%d", b.N)
+	errs := []error{}
+	b.StartTimer()
+	for n := 0; n < Iter; n++ {
+		// StartRC waits around for the replicas to get created
+		rc, err := framework.StartRC(string(data), uint(Parallelism), string(util.NewUUID()), m.RestClient)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		// StopRC waits around for the replicas to get deleted
+		err = framework.StopRC(rc, m.RestClient)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	b.StopTimer()
+	for err := range errs {
+		glog.Errorf("Unexpected error: %v", err)
+	}
+}
+
+// Benchmark pod listing by creating `Parallelism` pods, and listing them based on b.N
+func BenchmarkPodList(b *testing.B) {
+	b.StopTimer()
+	m := framework.NewMasterComponents(&framework.Config{nil, true})
+	defer m.Stop()
+
+	data, err := ioutil.ReadFile("benchmark-controller.json")
+	if err != nil {
+		glog.Fatalf("Unexpected error: %v", err)
+	}
+	framework.StartRC(string(data), uint(Parallelism), string(util.NewUUID()), m.RestClient)
```

Comment:
I was thinking of the shippable case, where we have like 10s to run a test and want to just skip to listing 100 pods or something meaningful instead of going 1, 10 etc. I think it'd be useful to time how long creating, listing, watching 100 pods takes for every pr, and I'd like to do this without sinking 10m per build (even if it is post build and doesn't hold up checkin). Thoughts?